libtor/src/lib.rs

```diff
@@ -393,6 +396,40 @@ impl Tor {
     }
 }
 
+/// Generate a hashed password to use HashedControlPassword
+pub fn generate_hashed_password(secret: &str) -> String {
+    // This code is rewrite of
+    // https://gist.github.com/s4w3d0ff/9d65ec5866d78842547183601b2fa4d5
+    // s4w3d0ff and jamesacampbell, Thank you!
+
+    let slen = 8 + (secret.len());
```

Comment:
Maybe here it's better to use `secret.as_bytes().len()`, because with unicode characters they can differ

Reply:
Alternatively, take a `&[u8]` instead of a `&str` for your secret

Reply:
I fixed.